tests/aggregation/tests.py

```diff
@@ -1340,3 +1344,198 @@ def test_aggregation_random_ordering(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age', default=0))
+                )
+                self.assertEqual(result['value'], 0)
+
+    def test_aggregation_default_integer(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age', default=21))
+                )
+                self.assertEqual(result['value'], 21)
+
+    def test_aggregation_default_expression(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age', default=Value(5) * Value(7)))
+                )
+                self.assertEqual(result['value'], 35)
+
+    def test_aggregation_default_group_by(self):
+        qs = (
+            Publisher.objects.values('name')
+            .annotate(books=Count('book'), pages=Sum('book__pages', default=0))
+            .filter(books=0)
+        )
+        self.assertEqual(
+            list(qs),
+            [{'name': "Jonno's House of Books", 'books': 0, 'pages': 0}],
+        )
+
+    def test_aggregation_default_complex(self):
+        # Scale rating to a percentage; default to 50% if no books published.
+        formula = Avg('book__rating', default=2.5) * 20.0
+        queryset = Publisher.objects.annotate(rating=formula)
+        self.assertQuerysetEqual(queryset, [
+            ('Apress', 85.0),
+            ("Jonno's House of Books", 50.0),
+            ('Morgan Kaufmann', 100.0),
+            ('Prentice Hall', 80.0),
+            ('Sams', 60.0),
+        ], lambda obj: (obj.name, obj.rating), ordered=False)
+
+    def test_aggregation_default_complex_time_python(self):
+        expr = Min(
+            'store__friday_night_closing',
+            filter=~Q(store__name='Amazon.com'),
+            default=datetime.time(17),
+        )
+        queryset = Book.objects.annotate(oldest_store_opening=expr)
+        self.assertQuerysetEqual(queryset, [
+            ('159059725', datetime.time(23, 59, 59)),
+            ('067232959', datetime.time(17)),
+            ('159059996', datetime.time(21, 30)),
+            ('013235613', datetime.time(21, 30)),
+            ('013790395', datetime.time(23, 59, 59)),
+            ('155860191', datetime.time(21, 30)),
+        ], lambda obj: (obj.isbn, obj.oldest_store_opening), ordered=False)
+
+    def test_aggregation_default_complex_time_database(self):
+        now = timezone.now()
+        if connection.vendor != 'postgresql':
+            now = now.astimezone(timezone.utc)
```

Comment:
For some reason PostgreSQL behaves differently to the other backends. I'm not sure why.

Is this sort of thing acceptable in tests?